enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,249 @@
+
+Jenkins itself is not native to Kubernetes and does not use a Java runtime that has been optimized for cloud-native environments (i.e. Quarkus).
+As a result, it has poor resilience when run on Kubernetes.
+The Jenkins templates do not run Jenkins in a highly available mode, leaving its primary instance vulnerable to downtime from slow restarts.
+The Jenkins container image is also very large - it is in fact the largest image in the OpenShift payload today.
+Removing Jenkins from the OpenShift payload has been a longstanding goal that can reduce the time needed to pull the OCP payload at install/updgrade time.
+
+Red Hat does not have a strong connection to the Jenkins community, and has marginal influence when it comes to requested features and patches.
+Recently Red Hat tried to engage with a third party to develop an open source Kubernetes operator for Jenkins.
+That effort unfortunately fell apart, leading Red Hat to fork the operator in an attempt to continue its development.
+No other companies or individuals have contributed to the operator since, and its current form does not provide measurable value beyond what is currently available with the Jenkins templates.
+
+With the upcoming GA release of OpenShift Pipelines, we would like to decouple Jenkins from OpenShift.
+This would allow bug fixes in our Jenkins distribution to be released on its own cadence.
+Customers can continue to deploy Jenkins on OpenShift via its ImageStream and associated templates.
+In this fashion, Jenkins becomes just another application on OpenShift and does not retain any preferential status.
+
+### Goals
+
+* Disable the integration points between Jenkins and OpenShift.
+* Move Jenkins out of the OCP Payload
+* Document how JenkinsPipeline strategy builds can be migrated off of OpenShift.
+
+### Non-Goals
+
+* Migrate Jenkins users to OpenShift Pipelines/Tekton.
+* Stop distributing the Jenkins image and our associated OpenShift plugins.
+
+## Proposal
+
+### User Stories
+
+As a developer using Jenkins on OpenShift
+I want to continue using my Jenkinsfiles
+So that I can continue to use Jenkins for my CI/CD processes
+
+As an OpenShift release engineer
+I want to remove Jenkins from the OCP payload
+So that I can reduce the size of the payload
+And release Jenkins fixes on their own cadence
+
+### Implementation Details/Notes/Constraints [optional]
+
+Deprecation warnings for the Jenkins Pipeline Build Strategy will need to be updated in the next OpenShift release (OCP 4.N).
+Since Jenkins integrations have been officially been officially deprecated for well over one year, these warnings need to be strengthened so customers are fully aware that all direct Jenkins integrations will be removed.
+Like the use of other deprecated APIs, OpenShift must also alert users if a Jenkins pipeline build strategy is used in a Build or BuildConfig.
+
+In the OCP 4.(N+1) release, builds which use the Jenkins pipeline build strategy should fail immediately.
```

Comment:
i don't think we have ever removed/disabled a v1 GA api.  i'm not sure who needs to sign off on this, but i think we'll need architecture team buy-in on whether this is allowed and if it is, what the process/mechanics need to be.

Reply:
We never removed anything, at most we've made some fields no-ops. I'd start with @smarterclayton @derekwaynecarr to confirm. 
Most importantly do we have any numbers behind this, how many customers are actually using this? How painful

Reply:
In a developer survey we did last year with customers, >40% of respondents were using Jenkins on OpenShift and every customer we know that uses Jenkins on OpenShift uses the pipeline build strategy. The impact would be high.

Reply:
Yeah, that I'd probably start simple with what @adambkaplan writes towards the end of this document, to just remove the jenkins image from the payload, and align that image with the rest of samples operator. But leaving the strategy as is. This will have some downsides for users, which for disconnected environments would need additional step during mirroring, but I think that would send a clear singal for future chanegs.